### DISCUSSION TITLE
[Python SDK] Link prompt to OpenAI Chat Completion directly

### DISCUSSION THREAD

**Discussion by fancyweb** [langfuse/langfuse#1096] [Ideas]
### Describe the feature or potential improvement

I didn't find any way to link an OpenAI Chat Completion generation to a Prompt. Judging from the code, it's not currently possible.

`prompt` should probably be allowed as an extra argument and correctly cascaded to the Generation span:
```python
my_prompt = langfuse.get_prompt('prompt1')
openai.chat.completions.create(prompt=my_prompt, ...)
```
So that the Generation span created can be linked to the Prompt.

To go a little bit further, what if the passed Prompt was compiled directly and added as a system message when it's a simple string, or as the whole `messages` arg when it's already a JSON array of messages? So that we don't have to do this:
```python
my_prompt = langfuse.get_prompt('prompt1')
openai.chat.completions.create(prompt=my_prompt, messages=[{'role': 'system', 'content': my_prompt.compile()}], ...)
```

### Contribute

- [ ] Yes, I can implement this and raise a PR

**Comment by marcklingen** [comment]
Thanks for sharing, fully agree on both points.

1) is a simple improvements, will try to ship this soon and let you know here

2) is more related to how to use and track compiled prompts well. Ideally capturing the inserted variables is super useful to build a way better prompt editor within the prompt management feature.

**Comment by fancyweb** [comment]
I'm closing this one since idea number 1 was implemented, and idea number 2 is now irrelevant since we have chat prompt type :smile:
